src/Analyzers/CSharp/Analyzers/UseCollectionExpression/UseCollectionExpressionHelpers.cs

```diff
@@ -72,43 +72,52 @@ internal static class UseCollectionExpressionHelpers
                 return false;
         }
 
-        // Looks good as something to replace.  Now check the semantics of making the replacement to see if there would
-        // any issues.  To keep things simple, all we do is replace the existing expression with the `[]` literal. This
-        // is an 'untyped' collection expression literal, so it tells us if the new code will have any issues moving to
-        // something untyped.  This will also tell us if we have any ambiguities (because there are multiple destination
-        // types that could accept the collection expression).
-        var speculationAnalyzer = new SpeculationAnalyzer(
-            topMostExpression,
-            s_emptyCollectionExpression,
-            semanticModel,
-            cancellationToken,
-            skipVerificationForReplacedNode,
-            failOnOverloadResolutionFailuresInOriginalCode: true);
-
-        if (speculationAnalyzer.ReplacementChangesSemantics())
-            return false;
+        // HACK: Workaround lack of compiler information for collection expression conversions with casts.
```

Comment:
>HACK [](http://example.com/codeflow?start=11&length=4)

Are we expecting to fix this in [https://github.com/dotnet/roslyn/issues/68826](https://github.com/dotnet/roslyn/issues/68826)? If so, let's reference the issue here instead.

Reply:
I'll let @CyrusNajmabadi answer that one, since this is part of his change.